### PR TITLE
Auth: Add SyncPermissions post auth hook

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
This PR adds a post auth hook to the AuthN service which enriches the current Identity (authn.Identity) with the current, logged in user's permissions from the database. Besides, in case AuthN service is enabled then it does disable the LoadPermissionsMiddleware.

**Why do we need this feature?**
Currently, the LoadPermissionsMiddleware runs on each request. But we would like to control if the permission should be loaded from the DB because in the future we are planning to load the permissions from a signed JWT token. 

**Who is this feature for?**

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by mgyongyosi** [comment]
@kalleep I'm wondering if there is another feature flag required or this hook can use the AuthNService flag? I think these changes can be controlled by the same feature flag. What do you think?

File: pkg/services/authn/authn.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ type ClientParams struct {
 	CacheAuthProxyKey string
 	// LookUpParams are the arguments used to look up the entity in the DB.
 	LookUpParams login.UserLookupParams
+	// SyncPermissionsFromDB ensure that permissions are loaded from DB and added to the identity
+	SyncPermissionsFromDB bool
```
Comment (kalleep) [inline pkg/services/authn/authn.go:57]: Nit: Rename to SyncPermissions

File: pkg/services/authn/authnimpl/service.go [hunk 2]
```diff
@@ -152,6 +152,10 @@ func ProvideService(
 		s.RegisterPostAuthHook(sync.ProvideOAuthTokenSync(oauthTokenService, sessionService).SyncOauthTokenHook, 60)
 	}
 
+	if features.IsEnabled(featuremgmt.FlagAuthnService) {
```
Comment (kalleep) [inline pkg/services/authn/authnimpl/service.go:155]: We don't need to check the flag here. The service are not used if this flag is not set

File: pkg/services/authn/authnimpl/sync/permission_sync.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/authn"
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
+
+var (
+	errSyncPermissionsFromDBForbidden = errutil.NewBase(errutil.StatusForbidden, "permissions_from_db.sync.forbidden")
+)
+
+func ProvidePermissionsFromDBSync(acService accesscontrol.Service) *PermissionsFromDBSync {
```
Comment (kalleep) [inline pkg/services/authn/authnimpl/sync/permission_sync.go:16]: Nit: Same here "FromDB" does not add anything in my opinion. I think PermissionsSync is fine

**Review by Jguer — APPROVED** [review]

File: pkg/services/authn/authnimpl/service.go [hunk 4]
```diff
@@ -152,6 +152,8 @@ func ProvideService(
 		s.RegisterPostAuthHook(sync.ProvideOAuthTokenSync(oauthTokenService, sessionService).SyncOauthTokenHook, 60)
 	}
 
+	s.RegisterPostAuthHook(sync.ProvidePermissionsSync(accessControlService).SyncPermissionsHook, 70)
```
Comment (kalleep) [inline pkg/services/authn/authnimpl/service.go:155]: I think we need to run the hook after FetchSyncedUserHook to make sure teams are loaded correctly

Comment (Jguer) [reply pkg/services/authn/authnimpl/service.go:155]: Confirmed, Teams do need to be loaded

Comment (Jguer) [reply pkg/services/authn/authnimpl/service.go:155]: (priority 110?)

**Review by kalleep — APPROVED** [review]

**Pull request merged**